product_operating_unit/models/product_template.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2019 brain-tec AG (http://www.braintec-group.com)
+# Copyright (C) 2019 Open Source Integrators
+# Copyright (C) 2019 Serpent Consulting Services
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    def _default_operating_unit_ids(self):
+        return [(6, 0,
+                 [self.env['res.users'].operating_unit_default_get(
+                     self.env.uid).id])]
+
+    operating_unit_ids = fields.Many2many(
+        'operating.unit',
+        'product_operating_unit_rel',
+        string='Operating Units',
+        default=_default_operating_unit_ids
+    )
+
+    @api.constrains('operating_unit_ids', 'company_id')
+    def _check_company_operating_unit(self):
+        for record in self:
+            if record.company_id and record.operating_unit_ids:
+                for operating_unit in record.operating_unit_ids:
+                    if record.company_id != operating_unit.company_id:
+                        raise ValidationError(
+                            _('Configuration error. The Company in the Product'
+                              ' and in the Operating Unit must be the same.')
+                        )
+
+    @api.multi
+    @api.constrains('operating_unit_ids', 'categ_id')
+    def _check_operating_unit(self):
+        for record in self:
+            if record.operating_unit_ids and record.operating_unit_ids != \
+                    record.categ_id.operating_unit_ids:
+                raise ValidationError(_(
+                    '''The Operating Unit in the Product and in the
+                    Category must be the same.'''))
```

Comment:
@nikul-serpentcs The OU don't need to be the same. The OU of the product needs to include the ones from the category.

Category 1: OU 1, OU 2
Product in Category 1: OU 1, OU 2, OU 3